src/Core/src/Handlers/Image/ImageHandler.Windows.cs

```diff
@@ -8,33 +9,152 @@ namespace Microsoft.Maui.Handlers
 {
 	public partial class ImageHandler : ViewHandler<IImage, Image>
 	{
+		/// <inheritdoc/>
 		protected override Image CreatePlatformView() => new Image();
 
+		/// <inheritdoc/>
 		protected override void DisconnectHandler(Image platformView)
 		{
 			base.DisconnectHandler(platformView);
 			SourceLoader.Reset();
 		}
 
+		/// <inheritdoc/>
 		public override bool NeedsContainer =>
 			VirtualView?.Background != null ||
+			VirtualView?.Aspect == Aspect.AspectFill ||
 			base.NeedsContainer;
 
+		/// <inheritdoc/>
+		protected override void SetupContainer()
+		{
+			base.SetupContainer();
+
+			// VerticalAlignment only works when the child's Height is Auto
+			PlatformView.Height = double.NaN;
+
+			MapHeight(this, VirtualView);
+			MapWidth(this, VirtualView);
+			MapMaximumHeight(this, VirtualView);
+			MapMaximumWidth(this, VirtualView);
+		}
+
+		/// <inheritdoc/>
+		protected override void RemoveContainer()
+		{
+			base.RemoveContainer();
+
+			MapHeight(this, VirtualView);
+			MapWidth(this, VirtualView);
+			MapMaximumHeight(this, VirtualView);
+			MapMaximumWidth(this, VirtualView);
+		}
+
+		/// <summary>
+		/// Maps the abstract <see cref="IView.Height"/> property to the platform-specific implementations.
+		/// </summary>
+		/// <param name="handler">The associated handler.</param>
+		/// <param name="view">The associated <see cref="Image"/> instance.</param>
+		public static void MapHeight(IImageHandler handler, IImage view)
+		{
+			// VerticalAlignment only works when the container's Height is set and the child's Height is Auto. The child's Height
+			// is set to Auto when the container is introduced
+			if (handler.ContainerView is FrameworkElement element)
+			{
+				element.Height = view.Height;
+			}
+
+			ViewHandler.MapHeight(handler, view);
+		}
+
+		/// <summary>
+		/// Maps the abstract <see cref="IView.Width"/> property to the platform-specific implementations.
+		/// </summary>
+		/// <param name="handler">The associated handler.</param>
+		/// <param name="view">The associated <see cref="Image"/> instance.</param>
+		public static void MapWidth(IImageHandler handler, IImage view)
+		{
+			if (handler.ContainerView is FrameworkElement element)
+			{
+				element.Width = view.Width;
```

Comment:
If just the height needs to be set, why set the width too?

Reply:
@japarson Should this be setting the `handler.PlatformView.Width` to `NaN`?

Reply:
I don't know. I'll try it.